### ISSUE TITLE
Porting Weights

### ISSUE THREAD

**Issue by dchall88** [rykov8/ssd_keras#18]
Thanks for the work you have done here. I agree that Caffe is painful to use.

A quick question. How did you port the weights from Caffe to Keras? Do you have code to do this?
The reason I ask is that I would like to port the coco trained weights from the original repo.

**Comment by rykov8** [comment]
@dchall88 Hi! I compiled the original SSD repo and then porting is rather straightforward, some sample code is available in [this gist](https://gist.github.com/rykov8/5fa4d2faff2e5e4704df54729bf57159). Probably, I missed some imports or other stuff, but the idea is clear, I think. One important notice is that in current version of `SSD` in this repo layers `pool6_mbox_loc_flat` and `pool6_mbox_conf_flat` are not the same as the original ones, that is described in #2. For these layers you just have to select the middle of the convolutional weights from the original ones. Moreover, the authors have changed the architecture a bit in newer versions of the paper, so, be careful to port the right weights.

**Comment by sruu72** [comment]
I ported to this new architecture, it seam that authors change pool6 to conv layer, but after that prediction is really bad. What I did, was replaced pool6 by:
`# Prediction from conv9_2
    num_priors = 4
    x = Convolution2D(num_priors * 4, 3, 3, border_mode='same',
                      name='conv9_2_mbox_loc')(net['conv9_2'])
    net['conv9_2_mbox_loc'] = x
    flatten = Flatten(name='conv9_2_mbox_loc_flat')
    net['conv9_2_mbox_loc_flat'] = flatten(net['conv9_2_mbox_loc'])
    name = 'conv9_2_mbox_conf'
    if num_classes != 21:
        name += '_{}'.format(num_classes)
    x = Convolution2D(num_priors * num_classes, 3, 3, border_mode='same',
                      name=name)(net['conv9_2'])
    net['conv9_2_mbox_conf'] = x
    flatten = Flatten(name='conv9_2_mbox_conf_flat')
    net['conv9_2_mbox_conf_flat'] = flatten(net['conv9_2_mbox_conf'])
    priorbox = PriorBox(img_size, 264.0, max_size=315.0, aspect_ratios=[2],
                        variances=[0.1, 0.1, 0.2, 0.2], step=300.0, offset=0.5,
                        name='conv9_2_mbox_priorbox')
    net['conv9_2_mbox_priorbox'] = priorbox(net['conv9_2'])`
Have you any idea what I missed?

**Comment by rykov8** [comment]
@sruu72 did you also port the weights for this architecture? If I remember correctly, for the latest version of SSD the authors also changed the way how prior boxes are generated, probably, this is the reason of bad predictions in your case.

**Comment by sruu72** [comment]
Yes I ported weights from caffe for this, I also added offset and step parameters to prior boxes (which seems be not so important).
But issue was that I did conversion on Windows, when I switched to Linux every things work perfect :) 
BTW As far I don't know difference between Win/Linux
